qfi_opt/examples/benchmark_opt_methods.py

```diff
@@ -108,48 +119,61 @@ def run_nlopt(obj, obj_params, num_params, x0, solver):
     N = 4
     G = spin_models.collective_op(spin_models.PAULI_Z, N) / (2 * N)
 
-    obj_params = {}
-    obj_params["N"] = N
-    obj_params["dissipation"] = 0
-    obj_params["G"] = G
-
-    max_evals = 100
-
-    for num_params in [4, 5]:
-        lb = np.zeros(num_params)
-        ub = np.ones(num_params)
-        # x0 = 0.5 * np.ones(num_params)  # This is an optimum for the num_params==4 problems
-        # np.random.seed(0)
-        np.random.seed(1)
-        x0 = np.random.uniform(lb, ub, num_params)
-
-        match num_params:
-            case 4:
-                models = ["simulate_OAT", "simulate_ising_chain", "simulate_XX_chain"]
-            case 5:
-                models = ["simulate_TAT", "simulate_local_TAT_chain"]
-
-        for model in models:
-            print(model)
-            obj = getattr(spin_models, model)
-
-            plt.figure()
-
-            for solver in ["LN_NELDERMEAD", "LN_BOBYQA", "ORBIT", "POUNDER"]:
-                filename = solver + "_" + model + ".txt"
-                if not os.path.exists(filename):
-                    global all_f
-                    all_f = []
-                    if solver in ["LN_NELDERMEAD", "LN_BOBYQA"]:
-                        run_nlopt(obj, obj_params, num_params, x0, solver)
-                    elif solver in ["ORBIT"]:
-                        run_orbit(obj, obj_params, num_params, x0)
-                    elif solver in ["POUNDER"]:
-                        run_pounder(obj, obj_params, num_params, x0)
-                    np.savetxt(filename, all_f)
-                else:
-                    all_f = np.loadtxt(filename)
-
-                plt.plot(all_f, label=filename)
-            plt.legend()
-            plt.savefig("Results_" + model + ".png", dpi=300)
+    for d in np.append([0], np.linspace(0.1, 5, 20)):
```

Comment:
nit: more descriptive variable name, so that it's easier to tell what later code is doing at a glance
```suggestion
    for dissipation_rate in np.append([0], np.linspace(0.1, 5, 20)):
```
Also: I ideally `N` and `G` would be something like `num_spins` and `observable`, bu tey don't really appear anywhere else so it's NBD.